presentation/components/src/presentation-components/DataProvidersFactory.ts

```diff
@@ -27,6 +30,7 @@ export interface DataProvidersFactoryProps {
  * specific use cases.
  *
  * @public
+ * @deprecated uses [[PresentationTableDataProvider]] which is deprecated
```

Comment:
Maybe deprecate just the `createSimilarInstancesTableDataProvider` method rather than the whole class?

Reply:
It has only that one public method so I think it makes sense deprecating whole class.

Reply:
Agreed - the new `Table` component in `itwin-ui` doesn't have a notion of a data provider, so it makes no sense to have a "data provider factory".